src/Plugin/CustomerController/AjaxLogin.php

```diff
@@ -0,0 +1,115 @@
+use Magento\Customer\Controller\Ajax\Login as CustomerAjaxLoginController;
+use Hyva\Checkout\Result\Json as JsonResult;
+use Magento\Framework\Controller\ResultInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Ajax login will authenticate user, merge carts and correctly set session.
+ * Doing the above using graphql is a pain actually where we need to authenticate
+ * user, merge carts and then finally fetch cart info. Still, will face session
+ * issues. So rely on ajax login is better here.
+ */
+class AjaxLogin
+{
+    /**
+     * @var \Hyva\Checkout\Config\Checkout
+     */
+    private $hyvaCheckoutConfig;
+
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var \Hyva\Checkout\Model\SectionData\LoginSectionDataServiceFactory
+     */
+    private LoginSectionDataServiceFactory $loginSectionDataServiceFactory;
+
+    /**
+     * AjaxLogin constructor.
+     *
+     * @param  \Psr\Log\LoggerInterface  $logger
+     * @param  \Hyva\Checkout\Config\Checkout  $hyvaCheckoutConfig
+     * @param  LoginSectionDataServiceFactory  $loginSectionDataServiceFactory
+     */
+    public function __construct(
+        LoggerInterface $logger,
+        HyvaCheckoutConfig $hyvaCheckoutConfig,
+        LoginSectionDataServiceFactory $loginSectionDataServiceFactory
+    ) {
+        $this->logger = $logger;
+        $this->hyvaCheckoutConfig = $hyvaCheckoutConfig;
+        $this->loginSectionDataServiceFactory = $loginSectionDataServiceFactory;
+    }
+
+    /**
+     * Include cart and customer information in the ajax login result
+     *
+     * This is needed in the react app side in order to fetch the customer cart
+     * info using graphql. Graphql customerCart query needs customer token id
+     * in order to correctly fetch the details.
+     *
+     * @param  \Magento\Customer\Controller\Ajax\Login  $ajaxLoginController
+     * @param  \Magento\Framework\Controller\ResultInterface  $result
+     * @return \Magento\Framework\Controller\ResultInterface
+     */
+    public function afterExecute(
+        CustomerAjaxLoginController $ajaxLoginController,
+        ResultInterface $result
+    ): ResultInterface {
+        if (!$this->canProceed($result)) {
+            return $result;
+        }
+
+        try {
+            /** @var \Hyva\Checkout\Model\SectionData\LoginSectionDataService $sectionDataService */
+            $sectionDataService = $this->loginSectionDataServiceFactory->create();
+            $response = $result->getData();
+            $response['data'] = $sectionDataService->getSectionData();
+
+            return $result->setData($response);
+        } catch (\Exception $exception) {
+            $this->logger->critical($exception);
+            return $result;
+        }
+    }
+
+    /**
+     * @param  \Magento\Framework\Controller\ResultInterface  $result
+     * @return bool
+     */
+    private function canProceed(ResultInterface $result): bool
+    {
```

Comment:
what if the controller is used somewhere outside the checkout? Could it break anything? Or is `data` a new field so that the change is backwards compatible?

Reply:
`data` is a new attribute I added. so it wont make any issue outside checkout. if `data` keyword does not make sense and if we need to use something else, then please suggest a keyword. Naming is the toughest part :)

Reply:
Good point, with better naming that would have been more clear. What about `login-section-data`?

Reply:
i feel `login-` not needed here. probably `sectionData` or `customerData` makes much more sense. camelCasing is better.

Reply:
Then maybe customerData to be consistent with Magento